spec/support/intercept.rb

```diff
@@ -31,13 +31,22 @@ def submit(obj)
     def start; end
     def stop; end
   end
+end
 
-  config.before :each, intercept: true do
-    @intercepted = Intercept.new
-    allow(ElasticAPM::Transport::Base).to receive(:new).and_return(@intercepted)
-  end
+RSpec.shared_context 'intercept', shared_contex: :metadata do
```

Comment:
TIL about `shared_context`

Reply:
missing a _t_ in the second `shared_contex:`

Reply:
A `shared_context` allows you to create a scope to include in a test context and only interact with the objects within that context. Previously, we were stepping around a context in the Rspec configuration and making a global change (with `allow_any_instance_of`) that was bleeding into other tests. The `NoMethodError` on the agent object was indicative of one test's cleanup getting in the way of another test's execution.
Keeping the mocking to a context and its scope should avoid these problems. In the proposed changes, there is an `agent` object local to the context that is mocked to return a `@intercepted` object when `#transport` is called. These changes to the `agent` object will not affect other tests so we shouldn't see the strange `NoMethodError` anymore.

All that said, the tests are still failing and I'm not 100% convinced this is the solution yet. It's just a pattern I've typically seen used with Rspec. I'll hopefully have it working soon! If not, it's back to the drawing board ✏️